--- PKGBUILD.orig	2023-05-06 13:14:21.510529144 +0100
+++ PKGBUILD	2023-05-06 13:20:59.342063870 +0100
@@ -23,7 +23,7 @@ makedepends=(
   graphviz
   imagemagick
   python-sphinx
-  texlive-latexextra
+  # texlive-latexextra
   xmlto
 )
 options=('!strip')
@@ -51,7 +51,34 @@ _make() {
 }
 
 prepare() {
-  cd $_srcname
+  # Out-of-tree module signing
+  #
+  ######################################################
+  # this is added at the start of prepare() & replaces #
+  # 'cd $_srcname'                                     #
+  ######################################################
+  # uncomment for linux-xanmod-cacule & some other AUR kernels
+  # to match the Package Maintainer's variable for the kernel
+  # sources directory.
+  #
+  # local _srcname=linux-${_major}
+
+  msg2 "Rebuilding local signing key..."
+
+  cp -rf /usr/src/certs-local ./
+  cd certs-local
+
+  msg2 "Updating kernel config with new key..."
+
+  # NB: config path must be quoted for file globbing to work
+  # some kernels have multiple config files (e.g linux-libre)
+  # to see configurable options run:
+  # /usr/src/certs-local/genkeys.py -h
+  ./genkeys.py -v --config '../config*'
+
+  cd ../$_srcname
+
+# cd $_srcname
 
   echo "Setting version..."
   echo "-$pkgrel" > localversion.10-pkgrel
@@ -79,7 +106,8 @@ prepare() {
 
 build() {
   cd $_srcname
-  _make htmldocs all
+#  _make htmldocs all
+  _make all
 }
 
 _package() {
@@ -202,6 +230,33 @@ _package-headers() {
   echo "Adding symlink..."
   mkdir -p "$pkgdir/usr/src"
   ln -sr "$builddir" "$pkgdir/usr/src/$pkgbase"
+
+  # Out-of-tree module signing
+  ##################################################
+  # this is added at the end of _package-headers() #
+  ##################################################
+  # This is run in the kernel source / build directory
+  #
+  # some AUR kernels may also need to set $builddir to match
+  # the Package Maintainer's variable for the build dir
+  # inside the package:
+  #
+  # local builddir="$pkgdir/usr/lib/modules/${_kernver}/build"
+
+  msg2 "Local Signing certs for out-of-tree modules..."
+
+  certs_local_src="../certs-local"
+  certs_local_dst="${builddir}/certs-local"
+
+  # install certificates
+  ${certs_local_src}/install-certs.py $certs_local_dst
+
+  # install dkms tools
+  dkms_src="$certs_local_src/dkms"
+  dkms_dst="${pkgdir}/etc/dkms"
+  mkdir -p $dkms_dst
+
+  rsync -a $dkms_src/{kernel-sign.conf,kernel-sign.sh} $dkms_dst/
 }
 
 _package-docs() {
@@ -226,7 +281,7 @@ _package-docs() {
 pkgname=(
   "$pkgbase"
   "$pkgbase-headers"
-  "$pkgbase-docs"
+#  "$pkgbase-docs"
 )
 for _p in "${pkgname[@]}"; do
   eval "package_$_p() {
