--- PKGBUILD.orig	2022-05-04 14:16:36.336143903 +0100
+++ PKGBUILD	2022-05-04 14:23:20.101037539 +0100
@@ -116,7 +116,34 @@ sha256sums=('9bbcd185b94436f9c8fe977fa0e
             'a0c3ecc39c58349e6bd1444d42f598ec189f53c267e80d48c66e945c91a7b831')
 
 prepare() {
-  cd linux-${pkgver}
+  # Out-of-tree module signing
+  #
+  ######################################################
+  # this is added at the start of prepare() & replaces #
+  # 'cd $_srcname'                                     #
+  ######################################################
+  # uncomment for linux-xanmod-cacule & some other AUR kernels
+  # to match the Package Maintainer's variable for the kernel
+  # sources directory.
+  #
+  # local _srcname=linux-${_major}
+
+  msg2 "Rebuilding local signing key..."
+
+  cp -rf /usr/src/certs-local ./
+  cd certs-local
+
+  msg2 "Updating kernel config with new key..."
+
+  # NB: config path must be quoted for file globbing to work
+  # some kernels have multiple config files (e.g linux-libre)
+  # to see configurable options run:
+  # /usr/src/certs-local/genkeys.py -h
+  ./genkeys.py -v --config '../config*'
+
+  cd ../linux-${pkgver}
+
+#  cd linux-${pkgver}
 
   msg2 "Setting version..."
   scripts/setlocalversion --save-scmversion
@@ -336,6 +363,33 @@ _package-headers() {
   echo "Adding symlink..."
   mkdir -p "$pkgdir/usr/src"
   ln -sr "$builddir" "$pkgdir/usr/src/$pkgbase"
+
+  # Out-of-tree module signing
+  ##################################################
+  # this is added at the end of _package-headers() #
+  ##################################################
+  # This is run in the kernel source / build directory
+  #
+  # some AUR kernels may also need to set $builddir to match
+  # the Package Maintainer's variable for the build dir
+  # inside the package:
+  #
+  # local builddir="$pkgdir/usr/lib/modules/${_kernver}/build"
+
+  msg2 "Local Signing certs for out-of-tree modules..."
+
+  certs_local_src="../certs-local"
+  certs_local_dst="${builddir}/certs-local"
+
+  # install certificates
+  ${certs_local_src}/install-certs.py $certs_local_dst
+
+  # install dkms tools
+  dkms_src="$certs_local_src/dkms"
+  dkms_dst="${pkgdir}/etc/dkms"
+  mkdir -p $dkms_dst
+
+  rsync -a $dkms_src/{kernel-sign.conf,kernel-sign.sh} $dkms_dst/
 }
 
 pkgname=("$pkgbase" "$pkgbase-headers")
